src/platform/unix.rs

```diff
@@ -274,6 +274,9 @@ pub trait WindowBuilderExtUnix {
 
     #[cfg(feature = "x11")]
     fn with_x11_screen(self, screen_id: i32) -> Self;
+    #[cfg(feature = "x11")]
+    /// Build window with X11's parent window. Only relevant on X11.
+    fn with_x11_parent(self, parent_id: usize) -> Self;
```

Comment:
One other point worth making: is it possible to use the `winit::WindowId` here?

Reply:
`usize` is likely the wrong type to use here. X11 window IDs are `u32`s, as far as I can tell (despite what `x11-dl`'s typedefs would have you believe).

As of #2351, we can make a `WindowId` from arbitrary `u64`s, so using it instead should be fine.

Reply:
> One other point worth making: is it possible to use the winit::WindowId here?

> As of https://github.com/rust-windowing/winit/pull/2351, we can make a WindowId from arbitrary u64s, so using it instead should be fine.

Good idea, thanks! I'll do that.